Clarify nextTick section of getting-started guide

skray opened the pull request:
Following up on #83, clarify getting started guide to explain that `nextTick` is not necessary when testing DOM updates, but it is still needed for operations that require advancing the event loop.

eddyerburgh commented on docs/en/guides/getting-started.md:
```diff
@@ -106,7 +106,9 @@ it('button click should increment the count', () => {
-To simplify usage, `vue-test-utils` applies all updates synchronously so you don't need to use `Vue.nextTick` in your tests.
+To simplify usage, `vue-test-utils` applies all updates synchronously so you don't need to use `Vue.nextTick` to test DOM updates in your tests.
+
+`nextTick` is still necessary when you need to explictly advance the event loop, for operations such as asynchronous callbacks or promise resolution.
```
Can you add it in a note. Like this:

```*Note: lorem ipsum ...*```

skray commented:
Ok, moved it into a note, and fixed what I thought was some awkward wording in my first pass. Let me know if you'd like any more/less explanation, examples, etc.
